src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -1259,10 +1259,116 @@ public void OnHeader(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
             }
         }
 
+        public void OnStaticIndexedHeader(int index)
+        {
+            Debug.Assert(index <= H2StaticTable.Count);
+
+            ref readonly var entry = ref H2StaticTable.Get(index - 1);
```

Comment:
This of-by-one fixup should be happening inside H2StaticTable.Get. https://github.com/dotnet/aspnetcore/blob/fa2d86e5b6b88dd8d7ec4b6de9ca5640e64ed76f/src/Shared/runtime/Http2/Hpack/H2StaticTable.cs#L13

Reply:
https://github.com/dotnet/aspnetcore/pull/24730#issuecomment-673722239